server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/QueryParser.java

```diff
@@ -0,0 +1,172 @@
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.hasAttachment;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.originMailboxes;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.recipients;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.sender;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.subject;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldParser.BOOLEAN_PARSER;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldParser.MAIL_ADDRESS_PARSER;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldParser.STRING_PARSER;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldParser.ZONED_DATE_TIME_PARSER;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.afterOrEquals;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.beforeOrEquals;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.contains;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.containsIgnoreCase;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.equals;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.equalsIgnoreCase;
+
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import javax.mail.internet.AddressException;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.vault.search.Criterion;
+import org.apache.james.vault.search.CriterionFactory;
+import org.apache.james.vault.search.Query;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableTable;
+
+class QueryParser {
+
+    static class QueryParserException extends Exception {
+        QueryParserException(String message) {
+            super(message);
+        }
+    }
+
+    enum FieldName {
+        deletionDate,
+        deliveryDate,
+        recipients,
+        sender,
+        hasAttachment,
+        originMailboxes,
+        subject;
+
+        static FieldName getField(String fieldNameString) throws QueryParserException {
+            return Stream.of(values())
+                .filter(fieldName -> fieldName.name().equals(fieldNameString))
+                .findFirst()
+                .orElseThrow(Throwing.supplier(() -> new QueryParserException("fieldName: '" + fieldNameString + "' is not supported")));
+        }
+    }
+
+    enum Operator {
+        equals,
+        equalsIgnoreCase,
+        contains,
+        containsIgnoreCase,
+        beforeOrEquals,
+        afterOrEquals;
+
+        static Operator getOperator(String operator) throws QueryParserException {
```

Comment:
Why not `operatorString` as above? This would allow a better name for the poor `op` variable...